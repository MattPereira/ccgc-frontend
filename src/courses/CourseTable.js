import React from "react";
import { Table } from "react-bootstrap";
import HolesRow from "../common/HolesRow";

import { v4 as uuidv4 } from "uuid";

/** Course table component.
 *
 * accepts pars and handicaps
 *
 * renders table with course data
 *
 */

const CourseTable = ({ pars, handicaps }) => {
  return (
    <Table responsive bordered hover className="text-center">
      <thead>
        <HolesRow />
      </thead>
      <tbody>
        <tr className="table-secondary">
          <th>PAR</th>
          {Object.values(pars).map((p) => (
            <td key={uuidv4()}>{p}</td>
          ))}
        </tr>

        <tr className="bg-white">
          <th>HCP</th>
          {Object.values(handicaps).map((h) => (
            <td key={uuidv4()}>{h}</td>
          ))}
          <td>--</td>
        </tr>
      </tbody>
    </Table>
  );
};

export default CourseTable;
