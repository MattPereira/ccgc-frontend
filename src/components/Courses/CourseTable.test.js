import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CourseTable from "./CourseTable";

const pars = {
  hole1: 3,
  hole2: 4,
  hole3: 5,
  hole4: 3,
  hole5: 4,
  hole6: 5,
  hole7: 3,
  hole8: 4,
  hole9: 5,
};

const handicaps = {
  hole1: 1,
  hole2: 2,
  hole3: 3,
  hole4: 4,
  hole5: 5,
  hole6: 6,
  hole7: 7,
  hole8: 8,
  hole9: 9,
};

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <CourseTable pars={pars} handicaps={handicaps} />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CourseTable pars={pars} handicaps={handicaps} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays table rows for holes, pars, and handicaps", function () {
  render(
    <MemoryRouter>
      <CourseTable pars={pars} handicaps={handicaps} />
    </MemoryRouter>
  );

  const holeRowHeader = screen.getByText("HOLE");
  expect(holeRowHeader).toBeInTheDocument();

  const parRowHeader = screen.getByText("PAR");
  expect(parRowHeader).toBeInTheDocument();

  const handicapRowHeader = screen.getByText("HCP");
  expect(handicapRowHeader).toBeInTheDocument();
});
