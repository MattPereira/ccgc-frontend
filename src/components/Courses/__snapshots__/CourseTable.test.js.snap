// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="text-center table table-bordered table-hover"
    >
      <thead>
        <tr
          class="table-dark"
        >
          <th>
            HOLE
          </th>
          <th>
            1
          </th>
          <th>
            2
          </th>
          <th>
            3
          </th>
          <th>
            4
          </th>
          <th>
            5
          </th>
          <th>
            6
          </th>
          <th>
            7
          </th>
          <th>
            8
          </th>
          <th>
            9
          </th>
          <th>
            10
          </th>
          <th>
            11
          </th>
          <th>
            12
          </th>
          <th>
            13
          </th>
          <th>
            14
          </th>
          <th>
            15
          </th>
          <th>
            16
          </th>
          <th>
            17
          </th>
          <th>
            18
          </th>
          <th>
            TOT
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="table-secondary"
        >
          <th>
            PAR
          </th>
          <td>
            3
          </td>
          <td>
            4
          </td>
          <td>
            5
          </td>
          <td>
            3
          </td>
          <td>
            4
          </td>
          <td>
            5
          </td>
          <td>
            3
          </td>
          <td>
            4
          </td>
          <td>
            5
          </td>
        </tr>
        <tr
          class="bg-white"
        >
          <th>
            HCP
          </th>
          <td>
            1
          </td>
          <td>
            2
          </td>
          <td>
            3
          </td>
          <td>
            4
          </td>
          <td>
            5
          </td>
          <td>
            6
          </td>
          <td>
            7
          </td>
          <td>
            8
          </td>
          <td>
            9
          </td>
          <td>
            --
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
